src/main.rs

```diff
@@ -765,12 +726,12 @@ async fn process_signed_delegate_action(
     }
 
     // Check if the SignedDelegateAction includes a FunctionCallAction that transfers FTs to BURN_ADDRESS
-    if USE_PAY_WITH_FT.clone() {
+    if *USE_PAY_WITH_FT {
         let non_delegate_actions = signed_delegate_action.delegate_action.get_actions();
         let treasury_payments: Vec<Action> = non_delegate_actions
-            .iter()
-            .filter_map(|action| {
-                if let Action::FunctionCall(FunctionCallAction { args, .. }) = action {
+            .into_iter()
```

Comment:
just curious, is the `.into_iter().filter(...)` better performance wise than `.iter().filter_map(...)` or is it stylistic?

Reply:
It removes the need to clone: `into_iter` returns an `Iterator<Item = T>` whereas `iter` returns an `Iterator<Item = &T>`. The switch from `filter_map` to `filter` is just because no actual mapping was being performed in the lambda.